File: ruby/resistor-color-duo/resistor_color_duo.rb
=begin
Write your code for the 'Resistor Color Duo' exercise in this file. Make the tests in
`resistor_color_duo_test.rb` pass.

To get started with TDD, see the `README.md` file in your
`ruby/resistor-color-duo` directory.
=end

module ResistorColorDuo
  
  RESISTOR_VALUE = ["black", "brown", "red", "orange", "yellow", "green", "blue", "violet", "grey", "white"]
  
  def self.value(color_array)
   
    color_array.map{|color| RESISTOR_VALUE.index(color)}.slice(0, 2).join.to_i

  end
  
end

File: javascript/resistor-color-duo/resistor-color-duo.js
//
// This is only a SKELETON file for the 'Resistor Color Duo' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export const decodedValue = (array) => {
  
  const COLOR_ARRAY = ["black", "brown", "red", "orange", "yellow", "green", "blue", "violet", "grey", "white"]
  
  return Number(array.slice(0, 2).map(color => COLOR_ARRAY.indexOf(color)).join(""))
  
};
